pageserver/src/layered_repository.rs

```diff
@@ -174,32 +270,45 @@ impl Repository for LayeredRepository {
         let metadata = TimelineMetadata::new(Lsn(0), None, None, Lsn(0), initdb_lsn, initdb_lsn);
         timeline::save_metadata(self.conf, timeline_id, self.tenant_id, &metadata, true)?;
 
+        let remote_client = if self.upload_layers {
+            let remote_client =
+                create_remote_timeline_client(self.conf, self.tenant_id, timeline_id)?;
```

Comment:
This way separate clients wont share concurrency limits. S3Bucket contains a semaphore inside to avoid hitting s3 limits on concurrent requests. So most probably we need to either have a global one or initialize it once like it was previously and explicitly pass it to repository constructor